fe/fe-core/src/main/java/com/starrocks/sql/optimizer/OptimizerContext.java

```diff
@@ -77,6 +79,9 @@ public class OptimizerContext {
     // uniquePartitionIdGenerator for external catalog
     private long uniquePartitionIdGenerator = 0L;
 
+    // collect all LogicalOlapScanOperators in the query before any optimization
+    private List<LogicalOlapScanOperator> allLogicalOlapScanOperators;
```

Comment:
why change it from OptimizerContext to TaskContext?

Reply:
Because text based rewrite is executed before `prepareMvRewrite` to avoid overhead cost, so needs to collect previous base tables before rewrite in `prepare`.

```
 public OptExpression optimize(ConnectContext connectContext,
                                  OptExpression logicOperatorTree,
                                  Map<Operator, ParseNode> optToAstMap,
                                  StatementBase stmt,
                                  PhysicalPropertySet requiredProperty,
                                  ColumnRefSet requiredColumns,
                                  ColumnRefFactory columnRefFactory) {
        // prepare for optimizer
        prepare(connectContext, columnRefFactory);

        // try text based mv rewrite first before mv rewrite prepare so can deduce mv prepare time if it can be rewritten.
        logicOperatorTree = new TextMatchBasedRewriteRule(connectContext, stmt, optToAstMap)
                .transform(logicOperatorTree, context).get(0);

        // prepare for mv rewrite
        prepareMvRewrite(connectContext, logicOperatorTree, columnRefFactory, requiredColumns);

        OptExpression result = optimizerConfig.isRuleBased() ?
                optimizeByRule(logicOperatorTree, requiredProperty, requiredColumns) :
                optimizeByCost(connectContext, logicOperatorTree, requiredProperty, requiredColumns);

        // clear caches in OptimizerContext
        context.clear();

        return result;
    }
```